### ISSUE TITLE
Remove deprecated methods

### ISSUE THREAD

**Issue by guillep** [pharo-project/pharo-vm#821]
Several methods call `self deprecated` and should be removed.

```smalltalk
objectArg: offset
	"Like #stackObjectValue: but access method arguments left-to-right"
	"Obsolete; was never used; replaced with methodReturnFloat: in the VirtualMachine struct."
	| oop |
	self deprecated.
	oop := self methodArg: offset.
	(self isImmediate: oop) ifTrue: [self primitiveFail. ^ nil].
	^oop
```

**Comment by singjyuwong** [comment]
Do you mean removing the methods or just removing `self deprecated`? Because `StackInterpreter >> objectArg:` and `InterpreterProxy >> objectArg:` don't have any senders, I think it means these two methods are not used.

**Comment by PalumboN** [comment]
Yes @kumom it refers to remove all the methods calling `self deprecated`. They should not have senders.

**Comment by singjyuwong** [comment]
I am a bit confused which branch I should base my changes on.

The following three deprecated methods in `pharo-10` are not deprecated or do not exist in `pharo-12`:
- `StackInterpreterSimulator >> byteAt:` (not deprecated)
- `StackInterpreterSimulator >> longAt:` (do not exist)
- `StackInterpreterSimulator >> longAt:put:` (do not exist)

What should I do with these three methods? In my current pharo image, I am basing my changes on version 12. Should I change it to version 10 instead?

**Comment by PalumboN** [comment]
It's ok, work on pharo-12 branch, that is the current development version.

Probably we already made some cleaning there.

**Comment by singjyuwong** [comment]
I don't see there are deprecated methods to remove on pharo-12 branch. For example, deprecated methods in pharo-10 like `InterpreterProxy >> floatArg:` have been removed.

**Comment by PalumboN** [comment]
Ok, so I think that we can close this issue! Thanks @kumom
